spec/controllers/api/v1/rents_controller_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+describe Api::V1::RentsController do
+  include_context 'with Authenticated User'
+
+  describe 'GET #index' do
+    subject(:http_request) { get :index, params: { user_id: user.id } }
+
+    context 'when fetching all the rents' do
+      let(:book) { create(:book) }
+      let(:rents) { create_list(:rent, 3, user_id: user.id, book_id: book.id, start_date: Time.zone.today, end_date: Date.tomorrow) }
+
+      it 'reponses with rents json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          rents, each_serializer: RentSerializer
+        ).to_json
+        expect(http_request.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(http_request).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    subject(:http_request) { post :create, params: { user_id: user.id, book_id: book_id, start_date: '2019-10-07', end_date: '2019-10-08' } }
+
+    context 'when request is valid' do
+      let(:book_id) { create(:book).id }
+      let(:rent) { create(:rent, user_id: user.id, book_id: book_id, start_date: '2019-10-07', end_date: '2019-10-08') }
```

Comment:
you must not create the rent record here, this is a false positive, because even if the http_request fails when you check the properties below the record always be created and always be truth

Reply:
Done